src/player.cpp

```diff
@@ -5313,7 +5313,7 @@ void player::suffer()
 
         if( in_sleep_state() && !has_effect( effect_narcosis ) ) {
             inventory map_inv;
-            map_inv.form_from_map( g->u.pos(), 2 );
+            map_inv.form_from_map( g->u.pos(), 2, dynamic_cast<player *>( &g->u ) );
```

Comment:
This cast shouldn't be necessary. `g->u` is a `avatar`, which is a `player`, so `&g->u` should be convertible to `player *`.